src/parser.c

```diff
@@ -1066,8 +1066,6 @@ enum parse_rv parse_CChainImportTransaction(struct CChainImportTransactionState
       switch (state->state) {
         case 0: // sourceChain
             CALL_SUBPARSER(bidState, blockchain_id_t);
-            if(memcmp(&network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->bidState.buf, sizeof(blockchain_id_t)))
-              REJECT("Invalid XChain ID");
```

Comment:
I think we should use `decode_chain_id` to make sure it's still a valid chain here (/ prompt the user to say which one?)

Reply:
We're trying to reduce prompts, so I'll add a `decode` here that just checks that `blockchain_id` is one of the valid blockchain ids

Reply:
Sounds good.